frontend/src/app/shared/services/api.service.spec.ts

```diff
@@ -0,0 +1,16 @@
+import { TestBed } from "@angular/core/testing";
+
+import { ApiService } from "./api.service";
+
+describe("ApiService", () => {
+    let service: ApiService;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({});
```

Comment:
It seems that you need to add `imports: [HttpClientTestingModule]` here... That makes the frontend tests running fine here.

Reply:
Good find!